engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1795,23 +1797,7 @@ public void validate() throws ConstraintViolationException
       }
     }
 
-    ValidationContext validatonContext = new ValidationContext();
-    for (OperatorMeta n: operators.values()) {
-      if (n.nindex == null) {
-        findStronglyConnected(n, validatonContext);
-      }
-    }
-    if (!validatonContext.invalidCycles.isEmpty()) {
-      throw new ValidationException("Loops in graph: " + validatonContext.invalidCycles);
-    }
-
-    List<List<String>> invalidDelays = new ArrayList<>();
-    for (OperatorMeta n : rootOperators) {
-      findInvalidDelays(n, invalidDelays, new Stack<OperatorMeta>());
-    }
-    if (!invalidDelays.isEmpty()) {
-      throw new ValidationException("Invalid delays in graph: " + invalidDelays);
-    }
+    validateDAGStructure();
```

Comment:
This validation is kept to validate the DAG during dynamic DAG modification at runtime thorugh cli.